crypto/fipsmodule/ml_dsa/ml_dsa_ref/sign.c

```diff
@@ -248,25 +249,28 @@ int ml_dsa_sign_internal(ml_dsa_params *params,
   ml_dsa_polyvecl_invntt_tomont(params, &z);
   ml_dsa_polyvecl_add(params, &z, &z, &y);
   ml_dsa_polyvecl_reduce(params, &z);
-  if(ml_dsa_polyvecl_chknorm(params, &z, params->gamma1 - params->beta)) {
-    goto rej;
-  }
+  z_invalid = ml_dsa_polyvecl_chknorm(params, &z, params->gamma1 - params->beta);
 
   /* FIPS 204: line 21 Check that subtracting cs2 does not change high bits of w and low bits
    * do not reveal secret information */
   ml_dsa_polyveck_pointwise_poly_montgomery(params, &h, &cp, &s2);
   ml_dsa_polyveck_invntt_tomont(params, &h);
   ml_dsa_polyveck_sub(params, &w0, &w0, &h);
   ml_dsa_polyveck_reduce(params, &w0);
-  if(ml_dsa_polyveck_chknorm(params, &w0, params->gamma2 - params->beta)) {
+  w0_invalid = ml_dsa_polyveck_chknorm(params, &w0, params->gamma2 - params->beta);
+
+  /* FIPS 204: Algorithm 7 line 23 - Reject if either check fails (constant-time to avoid leaking
+     which check failed) */
+  if(z_invalid | w0_invalid) {
```

Comment:
I think the compiler can turn this into non-CT code, please use an appropriate `constant_time_*` function to get the branch condition.

Reply:
Removed this check, added documentation and references to https://pq-crystals.org/dilithium/data/dilithium-specification-round3-20210208.pdf section 5.5